### PR TITLE
Allow Int64.MinValue as valid nativeint literal, which it is (fixes #9524)

### LINKED ISSUE #9524: Tiny bug, -9223372036854775808n is valid, but not allowed
The compiler will complain, saying it is out of the allowable range when trying to use a literal like this:

```f#
-9223372036854775808n
```

However, the value is actually within range, the `nativeint` has the same range as `Int64`, and this shows that a variable of that type can indeed hold such value:

```f#
> -9223372036854775807n - 1n;;
val it : nativeint = -9223372036854775808n
```

It's not a big thing, only when you want to compare the `IntPtr` with that value this would actually happen, but this is usually not used with static values.

Contrast this with that `Int64` allows one to write `-9223372036854775808L` and that `unativeint` can be written as `18446744073709551615un` without problem.

Also of note is that if you compile in x86, the max value would be `-2147483648n`, just saying that values "outside of the range" are valid when compiling.

If I can find where this is checked (probably an inclusive vs exclusive bug), I can submit a PR. I know it is really a tiny, tiny bug...

### PR DESCRIPTION
**Pull request by abelbraaksma**
Fixes #9524

This tiny, tiny PR makes `-9223372036854775808n` a valid literal, which is in line with `-9223372036854775808L` and basically all other boundary values of signed integers. The change follows the same logic as for signed integers, and adds a few tests (which only compile with this change, btw, not sure how to do that differently).

Before:
image

After (also showing out-of-range still works):
image

### CODE REVIEW

**Comment by abelbraaksma** [comment]
I see this as a little exercise with making changes to the lexing of parsing. I doubt anyone would ever be worried by the actual bug, nonetheless, it made a good case for trying this workflow.

Green and ready for review :)

**Comment by abelbraaksma** [comment]
Ping @kevinransom ;)

File: src/fsharp/lex.fsl [hunk 1]
```diff
@@ -405,9 +405,13 @@ rule token args skip = parse
        UINT64(n) } 
 
  | nativeint  
-     { try 
-           NATIVEINT(int64 (removeUnderscores (lexemeTrimRight lexbuf 1)))
-       with _ ->  fail args lexbuf (FSComp.SR.lexOutsideNativeSigned()) (NATIVEINT(0L)) } 
+     { let s = removeUnderscores (lexemeTrimRight lexbuf 1)
+       // Allow <max_nativeint+1> to parse as min_nativeint.  Stupid but allowed because we parse '-' as an operator. 
+       if s = "9223372036854775808" then NATIVEINT(-9223372036854775808L, true) else
```
Comment (KevinRansom) [inline src/fsharp/lex.fsl:410]: Magic values are less than ideal:

Perhaps precompute the values as static values somewhere maybe :
let minNative = System.Int64.MinValue
let minNativeWithoutSign =
    let s= minNative.ToString(System.Globalization.CultureInfo.InvariantCulture)
    s.Substring(1, s.Length - 1)

Comment (cartermp) [reply src/fsharp/lex.fsl:410]: Are these .NET calls available at this stage? looking at the surrounding code it also has magical values which would lead me to believe some things aren't available

Comment (KevinRansom) [reply src/fsharp/lex.fsl:410]: @cartermp , yes they are available.  They are values in mscorlib et al.  Indeed, magical values are everywhere in the compiler, that doesn't make them anything other than less than ideal.

Comment (abelbraaksma) [reply src/fsharp/lex.fsl:410]: @KevinRansom I can certainly update, I actually considered it, but chose the "so as neighboring code" approach. Ik not a fan of the magic values either, glad you brought it up :). 

I'll check globally if this can be fixed easily by using the predefined constants.

Comment (abelbraaksma) [reply src/fsharp/lex.fsl:410]: Is there a place for such globally used constants I should add this too?

Comment (KevinRansom) [reply src/fsharp/lex.fsl:410]: Just somewhere near the top of the file, they aren't really generally useful.

Comment (abelbraaksma) [reply src/fsharp/lex.fsl]: Done, not just for this one, but for all integers. @kevinransom, this is ready for rereview :).

**Comment by abelbraaksma** [comment]
> Please consider refactoring the magical values, otherwise looks good.

Done for all min/max ranges, and each of the four "bad" values, also added a bit of documentation comments to existing code.
